include/kodlab_mjbots_sdk/attitude.h

```diff
@@ -0,0 +1,334 @@
+/**
+ * @file attitude.h
+ * @author Ethan Musser (emusser@seas.upenn.edu)
+ * @brief Defines an attitude object for storing inertial parameters.
+ * @date 7/7/22
+ * 
+ * @copyright 2022 The Trustees of the University of Pennsylvania. All rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <Eigen/Geometry>
+#include "kodlab_mjbots_sdk/pi3hat.h"
+#include "kodlab_mjbots_sdk/common_header.h"
+#include "kodlab_mjbots_sdk/rotations.h"
+#include <iostream>
+#include <iomanip>
+
+namespace kodlab
+{
+
+/**
+ * @brief Attitude object storing inertial data.
+ * @tparam Scalar[optional] numeric type
+ */
+template<typename Scalar = float>
+class Attitude
+{
+
+private:
+  /**
+   * @brief Inputted attitude quaternion before world offset rotation
+   */
+  Eigen::Quaternion<Scalar> att_quat_raw_ = {1, 0, 0, 0};
+
+  /**
+   * @brief Attitude quaternion in world frame
+   */
+  Eigen::Quaternion<Scalar> att_quat_ = {1, 0, 0, 0};
+
+  /**
+   * @brief Attitude euler angles in world frame
+   * @note Always follow the default <b>Extrinsic X-Y-Z> convention
+   */
+  mutable kodlab::ValidatedCache<rotations::EulerAngles<Scalar>> att_euler_;
```

Comment:
maybe default to zeros and invalid

Reply:
Already defaults to invalid.  It could default to zeros for the invalid, cached data, or simply contain garbled data until valid data is provided.  Either is fine really.

Reply:
Resolved in ced2bfa83422a5806f829c57fa072a9ec6c14f7b